feature/foryou/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/foryou/ForYouViewModel.kt

```diff
@@ -109,6 +110,23 @@ class ForYouViewModel @Inject constructor(
                 initialValue = OnboardingUiState.Loading,
             )
 
+    private val _showNotificationPermission = MutableStateFlow(false)
+    val showNotificationPermission: StateFlow<Boolean> = _showNotificationPermission
+    private val _dialogState = MutableStateFlow(false)
+    val dialogState: StateFlow<Boolean> = _dialogState
+
+    fun toggleDialog() {
+        viewModelScope.launch {
+            _dialogState.emit(!_dialogState.value)
+        }
```

Comment:
_dialogState.update {} is simpler

Reply:
val showNotificationPermission = savedStateHandle.getStateFlow(SHOW_NOTIFICATION_PERMISSION, false)
    val dialogState = savedStateHandle.getStateFlow(DIALOG_STATE, false)
    fun toggleDialog() {
        savedStateHandle[SHOW_NOTIFICATION_PERMISSION] = !dialogState.value
    }
    fun showNotificationPermission() {
        savedStateHandle[DIALOG_STATE] = !showNotificationPermission.value
    }



you want something Like this